src/coupled_array.jl

```diff
@@ -0,0 +1,47 @@
+type CoupledArray{T,T2} <: AbstractArray{Float64,1}
+  u::T
+  u_control::T2
+  order::Bool
+end
+
+Base.length(A::CoupledArray) = length(A.u) + length(A.u_control)
+Base.size(A::CoupledArray) = (length(A),)
+function Base.getindex(A::CoupledArray,i::Int)
+  if A.order == true
+    i <= length(A.u) ? A.u[i] : A.u_control[i-length(A.u)]
+  else
+    i <= length(A.u) ? A.u_control[i] : A.u[i-length(A.u)]
+  end
+end
+
+function Base.getindex(A::CoupledArray,I...)
+  A[sub2ind(A.u,I...)]
+end
+
+function Base.getindex(A::CoupledArray,I::CartesianIndex{1})
```

Comment:
Actually, it is needed:
`Got an exception of type LoadError outside of a @test
  LoadError: MethodError: no method matching sub2ind(::Tuple{Base.OneTo{Int64}}, ::CartesianIndex{1})
  Closest candidates are:
    sub2ind(::AbstractArray{T,N}, ::Any...) at abstractarray.jl:1419
    sub2ind(::Tuple{Base.OneTo}, ::Integer) at abstractarray.jl:1439
    sub2ind(::Tuple{Base.OneTo}, ::Integer...) at abstractarray.jl:1438
    ...
   in macro expansion at ./broadcast.jl:127 [inlined]
   in macro expansion at ./simdloop.jl:73 [inlined]
   in macro expansion at ./broadcast.jl:123 [inlined]
   in _broadcast!(::Base.#abs, ::Array{Float64,1}, ::Tuple{Tuple{Bool}}, ::Tuple{Tuple{Int64}}, ::Tuple{DiffEqJump.CoupledArray{Array{Float64,1},Array{Float64,1}}}, ::Type{Val{1}}) at ./broadcast.jl:117
   in broadcast! at ./broadcast.jl:172 [inlined]
   in broadcast_t at ./broadcast.jl:228 [inlined]
   in broadcast at ./broadcast.jl:230 [inlined]
   in ode_determine_initdt`